bundles/org.openhab.binding.pirateweather/src/main/java/org/openhab/binding/pirateweather/internal/handler/PirateWeatherWeatherAndForecastHandler.java

```diff
@@ -0,0 +1,867 @@
+                }
+                if (newForecastDays < 2) {
+                    toBeRemovedChannels.addAll(removeChannelsOfGroup(CHANNEL_GROUP_FORECAST_TOMORROW));
+                }
+                for (int i = newForecastDays; i < forecastDays; ++i) {
+                    toBeRemovedChannels
+                            .addAll(removeChannelsOfGroup(CHANNEL_GROUP_DAILY_FORECAST_PREFIX + Integer.toString(i)));
+                }
+            } else {
+                if (forecastDays == 0 && newForecastDays > 0) {
+                    toBeAddedChannels.addAll(
+                            createChannelsForGroup(CHANNEL_GROUP_FORECAST_TODAY, CHANNEL_GROUP_TYPE_DAILY_FORECAST));
+                }
+                if (forecastDays <= 1 && newForecastDays > 1) {
+                    toBeAddedChannels.addAll(
+                            createChannelsForGroup(CHANNEL_GROUP_FORECAST_TOMORROW, CHANNEL_GROUP_TYPE_DAILY_FORECAST));
+                }
+                for (int i = (forecastDays < 2) ? 2 : forecastDays; i < newForecastDays; ++i) {
+                    toBeAddedChannels
+                            .addAll(createChannelsForGroup(CHANNEL_GROUP_DAILY_FORECAST_PREFIX + Integer.toString(i),
+                                    CHANNEL_GROUP_TYPE_DAILY_FORECAST));
+                }
+            }
+            forecastDays = newForecastDays;
+        }
+        if (numberOfAlerts != newNumberOfAlerts) {
+            logger.debug("Rebuilding alerts channel groups.");
+            if (numberOfAlerts > newNumberOfAlerts) {
+                for (int i = newNumberOfAlerts + 1; i <= numberOfAlerts; ++i) {
+                    toBeRemovedChannels
+                            .addAll(removeChannelsOfGroup(CHANNEL_GROUP_ALERTS_PREFIX + Integer.toString(i)));
+                }
+            } else {
+                for (int i = numberOfAlerts + 1; i <= newNumberOfAlerts; ++i) {
+                    toBeAddedChannels.addAll(createChannelsForGroup(CHANNEL_GROUP_ALERTS_PREFIX + Integer.toString(i),
+                            CHANNEL_GROUP_TYPE_ALERTS));
+                }
+            }
+            numberOfAlerts = newNumberOfAlerts;
+        }
+        ThingBuilder builder = editThing().withoutChannels(toBeRemovedChannels);
+        for (Channel channel : toBeAddedChannels) {
+            builder.withChannel(channel);
+        }
+        updateThing(builder.build());
+
+        Channel sunriseTriggerChannel = getThing().getChannel(TRIGGER_SUNRISE);
+        sunriseTriggerChannelConfig = (sunriseTriggerChannel == null) ? null
+                : sunriseTriggerChannel.getConfiguration().as(PirateWeatherChannelConfiguration.class);
+        Channel sunsetTriggerChannel = getThing().getChannel(TRIGGER_SUNSET);
+        sunsetTriggerChannelConfig = (sunsetTriggerChannel == null) ? null
+                : sunsetTriggerChannel.getConfiguration().as(PirateWeatherChannelConfiguration.class);
+
+        updateStatus(ThingStatus.UNKNOWN);
```

Comment:
Not sure if the thing turns online in all cases / sequences. As the `bridgeStatusChanged` might be called before this or after. It is not enforced anywhere.